libraries/script-engine/src/ScriptEngine.cpp

```diff
@@ -2363,30 +2361,57 @@ void ScriptEngine::entityScriptContentAvailable(const EntityItemID& entityID, co
         }
     }
     else {
-      // ENTITY SCRIPT WHITELIST STARTS HERE
+        // ENTITY SCRIPT WHITELIST STARTS HERE
+        auto nodeList = DependencyManager::get<NodeList>();
+        bool passList = false;  // assume unsafe
```

Comment:
since this whole code section below is about if we pass the whitelist or not, it would make less complexity if you add another function (maybe in its own class) for checking a URL.  That way you can simply return true or false out of the function instead of making all these boolean checks with `passList`

Reply:
This is a good idea, helps with making the codebase cleaner but also makes the whitelist more accessible. Will probably do this in a later PR. :)